app/models/patient.rb

```diff
@@ -142,6 +166,24 @@ def self.contacted_since(datetime)
       first_contacts: first_contact.length, pledges_sent: 20 }
   end
 
+  def self.to_csv
+    CSV.generate(encoding: 'utf-8') do |csv|
+      csv << CSV_EXPORT_FIELDS.keys # Header line
+      all.each do |patient|
+        csv << CSV_EXPORT_FIELDS.values.map do |field|
+          value = patient.public_send(field)
```

Comment:
Otherwise, yes, your understanding is correct!  But maybe it would be more readable as something like:
```
if value.is_a?(Array)
  value.reject(&:blank?).join(', ')
else
  value
end
```
What do you think?